src/awkward/operations/convert.py

```diff
@@ -232,6 +232,12 @@ def to_numpy(array, allow_missing=True):
             ]
         )
 
+    elif (
+        str(ak.operations.describe.type(array)) == "datetime64"
+        or str(ak.operations.describe.type(array)) == "timedelta64"
+    ):
+        return numpy.array([array[i] for i in range(len(array))])
```

Comment:
`numpy.array([array[i] for i in range(len(array))])` in vectorized form is `numpy.arange(len(array))` (and you have to specify the dtype for consistency on Windows).

But why is `ak.to_numpy` returning `[0, 1, 2, 3, ...]` for datetimes/timedeltas?

Reply:
Also, if you fix the string representation ("datetime" and "timedelta" are the Datashape words; they leave off the "64"), then the type-check will be broken: check the format for `startswith("m")`.

Reply:
done